test/functional/cursor.test.js

```diff
@@ -3901,6 +3901,60 @@ describe('Cursor', function () {
     });
   });
 
+  describe('Cursor forEach Error propagation', function () {
+    let configuration;
+    let client;
+    let cursor;
+
+    beforeEach(function () {
+      configuration = this.configuration;
+      client = configuration.newClient({ w: 1 }, { maxPoolSize: 1 });
+    });
+
+    afterEach(async function () {
+      await cursor.close();
+      await client.close();
+    });
+
+    // NODE-2035
+    it('should propagate error when exceptions are thrown from an awaited forEach call', function () {
+      const docs = [{ a: 1 }, { a: 2 }, { a: 3 }];
+      let collection;
+      return client
+        .connect()
+        .then(
+          () => {
+            let db = client.db(configuration.db);
+            collection = db.collection('cursor_session_tests2');
```

Comment:
since creating a collection is still part of the setup, I would suggest moving that into the before block, too; furthermore, I'm not sure how important it is for this collection to be unique for our use case; it seems that you are just trying to limit the number of returned documents to a known quantity, if that is accurate, you can use the same collection name as the others, and then just set a unique key and use the find filter to retrieve everything with that unique key set (e.g., `{ my_unique_test_key: 1 }, { my_unique_test_key: 2 }`); if it *is* important for the collection to be unique, then you'd want to use the before block to drop it and recreate it; otherwise you might still have left over data from previous test runs

I also agree with Eric's suggestion to use async await to simplify the flow of the code here and further down; you could use a combination of async await and catch to have something like `await cursor.forEach(() => throw err).catch(err => { // expect err.message; })` if the error is thrown synchronously, it would fail the test, and if the expect fails, it would also fail the test, but if the error is passed to the callback, things would resolve successfully